### DISCUSSION TITLE
Is there any way to mix an arbitrary wave file with midi to leverage ladspa effects.

### DISCUSSION THREAD

**Discussion by daveschere** [FluidSynth/fluidsynth#1202] [General]
I would like to be able to play arbitrary clips from a wave file  and send midi commands and have them go through the affects processing.  So it would look like noteon/noteoff only it would be play_clip and stop_playing_clip.

**Comment by derselbst** [comment]
This is not possible out of the box. You could probably write a custom soundfont loader via fluidsynth's API:
https://www.fluidsynth.org/api/group__soundfont__loader.html

Alternatively, put all these clips in a soundfont file and load that soundfont into fluidsynth.
